semantic_conventions/trace/general.yaml

```
@@ -48,31 +42,67 @@ groups:
       - id: peer.ip
         type: string
         brief: >
-          Remote address of the peer (dotted decimal for IPv4 or
+          Remote address of the logical peer (dotted decimal for IPv4 or
           [RFC5952](https://tools.ietf.org/html/rfc5952) for IPv6)
         examples: '127.0.0.1'
-      - id: peer.port
+      - id: sock.peer.name
+        type: string
+        brief: Remote socket peer name.
+        examples: proxy.example.com
+      - id: sock.peer.addr
+        type: string
+        brief: >
+          Remote socket peer address (IPv4 or IPv6 for internet protocols, path for local communication,
+           [etc](https://man7.org/linux/man-pages/man7/address_families.7.html)).
+        examples: ['127.0.0.1', '/tmp/mysql.sock' ]
+      - id: sock.peer.port
         type: int
-        brief: 'Remote port number.'
-        examples: [80, 8080, 443]
+        brief: Remote socket peer port (if defined for the address family).
+        examples: 16456
+      - id: sock.family
+        type:
+          allow_custom_values: true
+          members:
+            - id: inet
+              value: 'inet'
+              brief: "IPv4 address"
+            - id: inet6
+              value: 'inet6'
+              brief: "IPv6 address"
+            - id: unix
+              value: 'unix'
+              brief: "Unix domain socket path/"
+        requirement_level:
+          conditionally_required: if different than `inet` and if any of `net.sock.peer.addr` or `net.sock.host.addr` are set
```

I think this requirement is incompatible with the schema. Previously, we did not differentiate inet and inet6 in net.peer.ip. Now if the schema renames this to net.peer.sock.addr, and it contains an IPv6, then this would result in bad data.

I don't know how to handle this nicely, but some not so nice alternatives:
- Only have `inet`, rely on addr format to distinguish. If no addr is set, then we don't know.
- Add inet_any.
- Specify that if not set, inet6 vs inet must be determined by looking at the address

Maybe the last one would be quite alright:
```suggestion
          conditionally_required: if neither `inet` nor `inet6` and if any of `net.sock.peer.addr` or `net.sock.host.addr` are set.
```